### PR TITLE
Doc: Update timeline example

### PR DESCRIPTION
**Pull request by ImportanceOfBeingErnest**
## PR Summary

Update timeline example to fetch data from the github API. This will allow the example to update itself once new matplotlib versions are released.  
Unlike in its initial proposal in #11403, this is embaced in a `try.. except`, such that the doc build will not fail in case no internet connection can be established.
Also it now uses a `stem` (and is hence useful to show `stem` usage).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/lines_bars_and_markers/timeline.py [hunk 1]
```diff
@@ -15,57 +15,98 @@
...
+    url += '?per_page=100'
+    data = json.loads(urllib.request.urlopen(url, timeout=.4).read().decode())
+
+    dates = []
+    names = []
+    for item in data:
+        if 'rc' not in item['tag_name'] and 'b' not in item['tag_name']:
+            dates.append(item['published_at'].split("T")[0])
+            names.append(item['tag_name'])
+except:
```
Comment (timhoffm) [inline examples/lines_bars_and_markers/timeline.py]:
```suggestion
except Exception:
```

No bare except (see last paragraph of https://docs.python.org/2/howto/doanddont.html#except). Not that it really matters here, but we shouldn't use bad style.

File: examples/lines_bars_and_markers/timeline.py [hunk 2]
```diff
@@ -15,57 +15,98 @@
...
-# Set the xticks formatting
-# format xaxis with 3 month intervals
-ax.get_xaxis().set_major_locator(mdates.MonthLocator(interval=3))
+
+markerline, stemline, baseline = ax.stem(dates, levels,
+                                         linefmt="C3-", basefmt="k-",
+                                         use_line_collection=True)
+
+plt.setp(markerline, mec="k", mfc="w", zorder=3)
+markerline.set_ydata(np.zeros(len(dates)))
```
Comment (timhoffm) [inline examples/lines_bars_and_markers/timeline.py]: This is a hack on the original purpose of `stem`. Not sure I like it, but OTOH, the loop before wasn't good either. At least, this needs explanation.

**Comment by ImportanceOfBeingErnest** [comment]
@timhoffm Do the new changes reflect the points you raised? 

The docbuild fails for no apparent (to me) reason. Any idea what the problem is?

**Comment by timhoffm** [comment]
Doc failure might be due to the warnings to be fixed by #13743 which first needs #13745.

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.1.x

**Pull request merged**